### PR TITLE
Add STP port polling and device information

### PR DESCRIPTION
**Pull request by vitalisator**
+ new SQL table ports_stp
+ discovery and polling for STP/RSTP ports
+ port information under device STP tab

### CODE REVIEW

File: html/includes/print-stp-ports.inc.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+echo '<tr bgcolor="#ffffff"><th>Port</th><th>Priority</th><th>State</th><th>Enable</th><th>Path cost</th><th>Designated root</th><th>Designated cost</th><th>Designated bridge</th><th>Designated port</th><th>Fwd trasitions</th></tr>';
```
Comment (laf) [inline html/includes/print-stp-ports.inc.php]: Can you break this down into indented lines:

```html
<tr>
    <td></td>
</tr>
```

**Comment by laf** [comment]
I think we really could do with stp html page being converted to bootgrid. We merged the first one without it but all tables should be sortable and searchable.

**Comment by vitalisator** [comment]
still learning git ;-)
need some help to create bootgrid table with links, my version not work and is actually commented out.

**Comment by laf** [comment]
You need to create a file called html/includes/table/port-stp.inc.php

If you take a look at the syslog file in that directory it should give you an idea of how to structure it.

**Comment by vitalisator** [comment]
I added  html/includes/table/stp-ports.inc.php but the table data wont be load :-(
need a hint why

**Comment by laf** [comment]
https://github.com/librenms/librenms/compare/master...laf:stp

Grab the html/includes/common/stp-ports.inc.php and html/includes/table/stp-ports.inc.php files. Some small changes and fixes to get it working.

I think you need to tidy the table up though, it's really cramped:

image

Search doesn't work as you've not coded that in.

You will also need to update schema to 095.sql.

Once those have been changed then we are good for a merge, thanks very much :)

**Comment by laf** [comment]
@vitalisator You still need to change 094.sql to 095.sql: git mv sql-schema/094.sql sql-schema/095.sql should suffice.

**Comment by vitalisator** [comment]
@laf Thanks for all hints.
I will first fix search before...

**Comment by laf** [comment]
No probs, when do you think you will get it done as we have other PRs on hold until this is merged? I can either ask them to rebase the sql and use 095 or we can merge without search for now if you can add it later.

**Comment by vitalisator** [comment]
@laf If I sort by first column the table is the empty, no idea why :-(
Search is first deactivated (have no time for this at the moment)
The rest should by fine.

PS. Is AutoDeploy broken?

**Comment by laf** [comment]
@vitalisator You must not be passing the ORDER info on properly. You can use logfile() to print out debug info to your /opt/librenms/logs/librenms.log file to get a better idea of what queries are being run.

You've still got a merge conflict as you've left 094.sql in

**Comment by vitalisator** [comment]
@laf It should be fine now, please merge.

**Pull request merged**